File: synapse/examples/python/bob_agent.py
#!/usr/bin/env python3
"""bob_agent.py — Bob's chat agent"""

import asyncio
from synapse import connect

async def main():
    mesh = await connect("nats://localhost:4222")

    await mesh.register(
        name="Bob's Agent",
        description="Friendly chat agent",
        capabilities=["chat"],
        skills=[
            {"id": "chat", "name": "Chat", "description": "Chat with Bob"},
        ],
    )

    def chat_handler(payload, context):
        text = payload.get("input", {}).get("text", "")
        print(f"[Bob] Received: '{text}'")
        return {"text": f"Bob says: I got your message! You said '{text}'"}

    mesh.on_request("chat", chat_handler)

    print("Bob agent online, waiting for messages...")

    try:
        while True:
            await asyncio.sleep(1)
    except KeyboardInterrupt:
        await mesh.close()

if __name__ == "__main__":
    asyncio.run(main())

File: synapse/examples/docker/agents/bob-agent.ts
// agents/bob-agent.ts — Bob registers with a "chat" skill and responds to requests
import Synapse from "./synapse.js";

async function main() {
  const natsUrl = process.env.NATS_URL || "nats://localhost:4222";
  const mesh = await Synapse.connect(natsUrl);

  await mesh.register({
    name: process.env.AGENT_NAME || "Bob's Agent",
    description: "Friendly chat agent",
    capabilities: ["chat"],
    skills: [{ id: "chat", name: "Chat", description: "Chat with Bob" }],
  });

  mesh.onRequest("chat", (payload) => {
    const text = payload.input?.text;
    console.log(`[Bob] Received: "${text}"`);
    return { text: `Bob says: Got your message! You said "${text}"` };
  });

  console.log("Bob agent online, waiting for messages...");
  process.on("SIGINT", async () => { await mesh.close(); process.exit(0); });
}

main().catch(console.error);
